src/sortStudents.ts

```diff
@@ -1,16 +1,69 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married ='married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+  const orderFactor = (order === 'asc') ? 1 : -1;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      sortedStudents.sort(
+        (a: Student, b: Student) => a.name.localeCompare(b.name),
+      );
+      break;
+
+    case SortType.Surname:
+      return sortedStudents.sort(
+        (a: Student, b: Student) => a.surname.localeCompare(b.surname),
+      );
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return sortedStudents.sort(
        (a: Student, b: Student) => a[sortBy].localeCompare(b[sortBy]),
      );
```
checklist p.3 [CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.